src/coreclr/jit/llvm.cpp

```diff
@@ -1896,7 +1802,252 @@ void Llvm::ConvertShadowStackLocalNode(GenTreeLclVarCommon* node)
+        }
+        else
+        {
+            indirNode = _compiler->gtNewOperNode(GT_IND, callReturnType, returnAddrLclAfterCall);
+        }
+        indirNode->gtFlags |= GTF_IND_TGT_NOT_HEAP; // No RhpAssignRef required
+        LIR::Use callUse;
+        if (CurrentRange().TryGetUse(callNode, &callUse))
+        {
+            callUse.ReplaceWith(_compiler, indirNode);
+        }
+        else
+        {
+            callNode->ClearUnusedValue();
+        }
+
+        GenTreePutArgType* putArg = _compiler->gtNewPutArgType(TYP_I_IMPL, returnAddrLcl, CORINFO_TYPE_PTR, nullptr);
+        lastArg = _compiler->gtInsertNewCallArgAfter(putArg, insertAfterArg);
+
+        callNode->gtReturnType = TYP_VOID;
+        callNode->gtCorInfoType = CORINFO_TYPE_VOID;
+        callNode->ChangeType(TYP_VOID);
+
+        CurrentRange().InsertBefore(callNode, shadowStackVar, offset, returnValueAddress, addrStore);
+        CurrentRange().InsertAfter(addrStore, returnAddrLcl, putArg);
+        CurrentRange().InsertAfter(callNode, returnAddrLclAfterCall, indirNode);
+    }
+    else
+    {
+        callNode->gtCorInfoType = calleeSigInfo.retType;
+    }
+    return lastArg;
+}
+
+void Llvm::failUnsupportedCalls(GenTreeCall* callNode, CORINFO_SIG_INFO &calleeSigInfo)
+{
+    // we can't do these yet
+    if (callNode->gtCallType == CT_INDIRECT || _isRuntimeImport(_thisPtr, callNode->gtCallMethHnd))
+    {
+        failFunctionCompilation();
+    }
+
+    // TODO-LLVM: not attempting to compile generic signatures with context arg via clrjit yet
+    if (calleeSigInfo.hasTypeArg())
+    {
+        failFunctionCompilation();
+    }
+
+    // TODO-LLVM: VSD calls 
+    // this call has 3 args, but only the struct RuntimeTypeHandle in the siginfo
+    //             RuntimeTypeHandle handle = instance.GetInterfaceImplementation(new RuntimeTypeHandle(new
+    //             EETypePtr(interfaceType)));
+    //
+    // / --*t2 ref this in rcx
+    // + --*t19 int arg2 in rdx
+    // +--*t137 int arg1 in r10                  N007(27, 18)[000020]-- CXG-- -----t20 =
+    // *CALLV stub struct System.Runtime.InteropServices.IDynamicInterfaceCastable.GetInterfaceImplementation
+    if (callNode->IsVirtual())
+    {
+        // The 3rd argument in this example is the address hidden argument VSD uses and always TYP_I_IMPL
+        failFunctionCompilation();
+    }
+
+    if (callNode->gtCallArgs != nullptr)
+    {
+        for (GenTree* operand : callNode->Operands())
+        {
+            if (operand->IsArgPlaceHolderNode() || !operand->IsValue())
+            {
+                // Either of these situations may happen with calls.
+                continue;
+            }
+            fgArgTabEntry* curArgTabEntry = _compiler->gtArgEntryByNode(callNode, operand);
+            regNumber      argReg         = curArgTabEntry->GetRegNum();
+            if (argReg == REG_STK || curArgTabEntry->argType == TYP_BYREF) // TODO-LLVM: out and ref args
```

Comment:
I wonder if things will "just work" if you remove the `argReg == REG_STK` restriction.